File: ABC/<300/270/Julia/B.jl
function main()
  X, Y, Z = parse.(Int, readline()|>split)
  m = findmin([X,Y,Z])[1]
  if Y<0
      X *= -1
      Y *= -1
      Z *= -1
  end
  if X < Y
      abs(X) |> println
  elseif Z > Y
      -1 |> println
  else
      if Z < 0
          -2Z + X |> println
      else
          X |> println
      end
  end
  return 0
end
main()

File: ABC/<300/270/C++/B.cpp
#include<iostream>
using namespace std;

int main(){
  int X, Y, Z; cin >> X >> Y >> Z;
  if(Y<0) X *= -1, Y *= -1, Z *= -1;
  if(X < Y) cout << abs(X) << endl;
  else if(Z > Y) cout << -1 << endl;
  else{
    if(Z < 0) cout << X-2*Z << endl;
    else cout << X << endl;
  }
  return 0;
}